Update README.md (#138)
purduesigbots/pros-docs: README.md

## How are these docs built?

> 

Reviewer: while this may be discussed later as well, I think it's still relevant information in this section and should not be removed.

Author: Ok. I agree with you. I'll move those back.

  in the API docs and tutorials. This is also modified from its original state, and that
  fork can be found at [this fork](https://github.com/purduesigbots/sphinx-tabs).

## I Just Want to Help
Great! Please help us. Choose one of the following that best suits your needs. 

> #### Easy

Author: For the easy/medium/hard distinction I just came up with something that sounded good. I'm not sold on that *wording* per say, but I do like having different methods that people can use.

> #### Easy

Author: I'm not entirely sure what you mean by having the sections. Would you like to elaborate on that a little?

> #### Easy

Reviewer: more like what was already in what is now the "hard" section-- section titles that describe a certain kind of change as opposed to how hard that change may be to make

> #### Easy

Author: I like that idea. I just commited a version of the README that rather than: easy, medium, and hard, they are Simple Text Changes, Formatting Changes, and Theme Changes.

> If you want ot change a page or a couple words, it is simple. All you have to do is fork this repository and change a `.rst` file. Then submit a pull request, or a PR, and you have officially contributed to `pros-docs`! 

Author: I'm about to commit a version of that with bullet points from line 24 - 26. Check those out and tell me what you think.

> If you want ot change a page or a couple words, it is simple. All you have to do is to edit a file and [Github's UI](https://help.github.com/en/articles/editing-files-in-another-users-repository) will take care of the rest of it! Then you have officially contributed to `pros-docs`! 

Author: I'm going to mention that. Also refer to the comment before this.

#### Medium

> Here you will install a basic html version of `pros-docs` This is adequate for most people

Author: I got that one

To make this process easier. If you don't already have a basic understanding of how these work:       [git](https://rogerdudler.github.io/git-guide/), 
                  [github](https://github.com/firstcontributions/first-contributions), 
                  [sphinx](https://www.mageworx.com/blog/2016/05/sphinx-the-beginners-guide/), 
                  [python3](https://wiki.python.org/moin/BeginnersGuide), you should go check them out before you start.
 
 Clone the project, change the directory to `pros-docs`
 ```sh

> git clone https://github.com/purduesigbots/pros-docs

Reviewer: you can make the next steps regarding submodules by running `git clone --recursive ...`. if someone forgets to add the flag when cloning, they can `git submodule update --init --recursive` to do all the rest in one step (and `git submodule update --recursive` to update later)

Author: I added a note below that code. Let me know if that solves it.

cd pros-docs
```

The `sphinx_rtd_theme` and `sphinx-tabs` folders within the documentation are
[git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). As a result,
you need to initialize these submodules and pull them separately when updating the docs.

```sh
# Repeat this in both the sphinx_rtd_theme and sphinx-tabs directories
> git submodule init
> git submodule update
```

 Install These Dependancies:
* Python3
* Sphinx (pip package)
* Sphinx-tabs (pip package)
* Ablog (pip package)

We recommend working in a virtual environment, but the following 2 lines are optional
> python3 -m venv ./docs-venv
> . ./docs-venv/bin/activate

> Whether or not you decide to use the python virtual environment install all of the dependencies for building the PROS Documentation, run:

Reviewer: ```suggestion
Whether or not you decide to use a virtual environment install all of the dependencies for building the PROS Documentation, run:
```

```sh
pip3 install -r requirements.txt
```
Note: to get out of the virtual environment run `deactivate`

Once you have installed the requirements, building is very straightforward. Just
run `make` in the `pros-docs` directory.

Now, pull up a browser and open `pros-docs/build/index.html`

> Great! Now you can edit the `.rst` files, make new files, and send a PR to change those files on the actual website.

Reviewer: not sure if it's just me, but this line sounds a little patronizing

Author: I think you're right. I didn't even think about that lol.

 
 Clone the project, change the directory to `pros-docs`
 ```sh
git clone --recursive https://github.com/purduesigbots/pros-docs
cd pros-docs
```
*Note: If you forget to add the `--recursive` flag, simply run `git submodule update --init --recursive`. Later on, if your submodules are out of date, you can run `git submodule update --recursive`*

The `sphinx_rtd_theme` and `sphinx-tabs` folders within the documentation are
[git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). As a result,
you need to initialize these submodules and pull them separately when updating the docs.

```sh
# Repeat this in both the sphinx_rtd_theme and sphinx-tabs directories
> git submodule init
> git submodule update
```

 Install These Dependancies:
* Python3
* Sphinx (pip package)
* Sphinx-tabs (pip package)
* Ablog (pip package)

```
 We recommend working in a virtual environment, but the following 2 lines are optional
> python3 -m venv ./docs-venv
> . ./docs-venv/bin/activate
```

> Whether or not you decide to use a virtual environment install all of the dependencies for building the PROS Documentation, run:

Reviewer: ```suggestion
Whether or not you decide to use a virtual environment, install all of the dependencies for building the PROS Documentation and then run:
```

```sh
pip3 install -r requirements.txt
```

> Note: to get out of the virtual environment run `deactivate`

Reviewer: be consistent with note styling

## I Just Want to Help
Great! Please help us. Choose one of the following that best suits your needs. 

#### Simple Text Changes
To change a page or a couple words, it is simple. You can:
 - Edit a file in this repository and [Github's UI](https://help.github.com/en/articles/editing-files-in-another-users-repository) will take care of the rest of it!
 - Go to the pros [website](https://pros.cs.purdue.edu/v5/okapi/tutorials/walkthrough/clawbot.html), and on the top right of each page is a `Edit on Github` button. Pressing that will also allow you to edit the source file.

#### Formatting Changes

> Here you will install a basic html version of `pros-docs` This is adequate for most formatting changes people.

Reviewer: ```suggestion
Here you will install a basic html version of `pros-docs` This is adequate for most people who are looking to change or fix formatting.
```

Great! Please help us. Choose one of the following that best suits your needs. 

#### Easy

> If you want ot change a page or a couple words, it is simple. All you have to do is to edit a file and [Github's UI](https://help.github.com/en/articles/editing-files-in-another-users-repository) will take care of the rest of it! Then you have officially contributed to `pros-docs`! 

Reviewer 2: you spell "to" "ot"

Now you can easily edit the `.rst`, and make new files. If you would like to, submit a PR to get those files changed on the website.

#### Theme Changes

> For more advanced users, if there is a bug in the theme, or *syntax highlighting*, for example. The tabs theming may be changed if you follow this guide. 

Reviewer: hmm, i think it's better to just remove references to level of difficulty entirely, and that would include phrases like "advanced users."

~also, is there a reason "syntax highlighting" is italicized?~ `theme, or *syntax highlighting*` -> `theme (like the syntax highlighting, for instance)`

lastly, I don't think we need to specifically call out the tabs theme here, as "theme" should cover both. the distinction can be left to later where it matters
